### PR TITLE
Add form verification example

### PR DESCRIPTION
**Pull request by cvializ**
/to @aghassemi 

Fixes ampproject/amphtml#9700

### CODE REVIEW

File: src/20_Components/amp-form.html [hunk 1]
```diff
@@ -205,6 +205,46 @@
...
+  If an error occurs, such as a username being taken, it will show as a
+  validation error in the form.
+
+  Read [here](https://www.ampproject.org/docs/reference/components/amp-form)
+  for complete documentation.
+
+  Use the `verify-xhr` attribute to enable the form verification feature.
+
+  Form verification can combine with custom error reporting using
+  `visible-when-invalid="customError"`.
```
Comment (sebastianbenz) [inline src/20_Components/amp-form.html]: How does this work? I couldn't find anything in the documentation either.

File: src/20_Components/amp-form.html [hunk 2]
```diff
@@ -205,6 +205,46 @@
...
+      <span visible-when-invalid="patternMismatch" validation-for="verification-username">Invalid character in username</span>
+    </div>
+    <input type="submit" value="Check" class="ampstart-btn caps">
+    <div submit-success>
+      <template type="amp-mustache">
+        Success! Thanks {{name}} for trying the <code>amp-form</code> demo! Try to insert the word "error" as a name input in the form to see how <code>amp-form</code> handles errors.
+      </template>
+    </div>
+    <div submit-error>
+      <template type="amp-mustache">
```
Comment (sebastianbenz) [inline src/20_Components/amp-form.html]: I'd be nice to have a sample here how to print the verification errors similar to the sample in the official docs.

**Comment by cvializ** [comment]
I'd be glad to! I was going for the level of brevity in the neighboring sections, but I understand if more detail is needed for a more complex feature.

File: src/20_Components/amp-form.html (hunk 1 above)
Comment (cvializ) [inline src/20_Components/amp-form.html]: `customError` is part of the [HTML5 Constraint Validation spec](https://dev.w3.org/html5/spec-preview/constraints.html#the-constraint-validation-api), like `patternMismatch` or  `valueMissing`. AMP uses `setCustomValidity` for the verification errors, which applies the `customError` validity to an element.

File: src/20_Components/amp-form.html (hunk 2 above)
Comment (cvializ) [inline src/20_Components/amp-form.html]: Done.

**Comment by cvializ** [comment]
PTAL

**Comment by cvializ** [comment]
Friendly ping @sebastianbenz  : )

**Comment by cvializ** [comment]
All good, thank you!

**Pull request merged**